src/mlpack/methods/ann/layer/softmin.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file methods/ann/layer/softmin.hpp
+ * @author Aakash Kaushik
+ *
+ * Definition of the Softmin class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTMIN_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTMIN_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+* Implementation of the Softmin layer. The Softmin function takes as a input
+* a vector of K real numbers, rescaling them so that the elements of the
+* K-dimensional output vector lie in the range [0, 1] and sum to 1.
+* 
+* @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+*         arma::sp_mat or arma::cube).
+* @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+*         arma::sp_mat or arma::cube).
+*/
```

Comment:
```suggestion
*  Implementation of the Softmin layer. The Softmin function takes as a input
*  a vector of K real numbers, rescaling them so that the elements of the
*  K-dimensional output vector lie in the range [0, 1] and sum to 1.
* 
*  @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
*          arma::sp_mat or arma::cube).
*  @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
*          arma::sp_mat or arma::cube).
*/
```

Should align with the first line.